go/adapter/opentelemetry/adapter.go

```diff
@@ -0,0 +1,148 @@
+package opentelemetry
+
+import (
+	"context"
+	"log"
+	"time"
+
+	observe "github.com/dylibso/observe-sdk/go"
+	trace "go.opentelemetry.io/proto/otlp/trace/v1"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+)
+
+type OTLPProtocol string
+
+const (
+	GRPC OTLPProtocol = "grpc"
+	HTTP OTLPProtocol = "http/protobuf"
+)
```

Comment:
these are the possible values we can support when passed into the config either directly as literals, as the `const`s here, or via an env var like `OTEL_EXPORTER_OTLP_PROTOCOL`